File: solutions/c/resistor-color/3/resistor_color.c
#include "resistor_color.h"

unsigned int color_code(resistor_band_t color) {
    return (unsigned int) color;
}

const resistor_band_t *colors() {
    static const resistor_band_t COLORS[10] = {
        0, 1, 2, 3, 4, 5, 6, 7, 8, 9
    };
    return COLORS;
}

File: solutions/java/resistor-color/3/src/main/java/ResistorColor.java
import java.util.Arrays;
import java.util.List;

class ResistorColor {
    static final private List<String> COLORS = List.of(
        new String[]{
            "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"
        }
    );
    
    static int colorCode(String color) {
        return COLORS.indexOf(color);
    }

    static String[] colors() {
        return COLORS.toArray(new String[0]);
    }
}
